tests/Integration/Behaviour/Features/Scenario/Product/Shop/update_multishop_options.feature

```diff
@@ -0,0 +1,104 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s product --tags update-multi-shop-options
+@restore-products-before-feature
+@clear-cache-before-feature
+@restore-shops-after-feature
+@restore-languages-after-feature
+@clear-cache-after-feature
+@product-multi-shop
+@update-multi-shop-options
+Feature: Feature: Update product options from Back Office (BO) for multiple shops
+  As a BO user
+  I want to be able to update product fields associated with options in multiple shops.
+
+  Background:
+    Given shop "shop1" with name "test_shop" exists
+    And manufacturer studioDesign named "Studio Design" exists
+    And manufacturer graphicCorner named "Graphic Corner" exists
+    And shop group "default_shop_group" with name "Default" exists
+    And I add a shop "shop2" with name "default_shop_group" and color "red" for the group "default_shop_group"
+    And I add a shop group "test_second_shop_group" with name "Test second shop group" and color "green"
+    And I add a shop "shop3" with name "test_third_shop" and color "blue" for the group "test_second_shop_group"
+    And I add a shop "shop4" with name "test_shop_without_url" and color "blue" for the group "test_second_shop_group"
+    And single shop context is loaded
+    And language "french" with locale "fr-FR" exists
+    Given I add product "product1" with following information:
+      | name[en-US] | Presta camera |
+      | type        | standard      |
+    And product "product1" should have following options:
+      | product option      | value |
+      | visibility          | both  |
+      | available_for_order | true  |
+      | online_only         | false |
+      | show_price          | true  |
+      | condition           | new   |
+      | show_condition      | false |
+      | manufacturer        |       |
+    When I update product "product1" options with following values:
+      | visibility          | catalog      |
+      | available_for_order | false        |
+      | online_only         | true         |
+      | show_price          | false        |
+      | condition           | used         |
+      | show_condition      | true         |
+      | manufacturer        | studioDesign |
+    And I copy product product1 from shop shop1 to shop shop2
+    Then product "product1" should have following options for shops "shop1,shop2":
+      | product option      | value        |
+      | visibility          | catalog      |
+      | available_for_order | false        |
+      | online_only         | true         |
+      | show_price          | false        |
+      | condition           | used         |
+      | show_condition      | true         |
+      | manufacturer        | studioDesign |
+    And product "product1" should not be indexed
```

Comment:
I think we need to be able to detect the indexation on a multishop level Since `Product::visibility` and `Product::indexed` are both multishop fields it seems like a product can be indexed in a shop but not in another one So we need to check this on a shop per shop basis like the other options assertions

I check the `ProductIndexationUpdater` for this, it relies on `Search::indexation` method which apparently handles multi shop data automatically internally So it seems no modification would be needed in the `ProductIndexationUpdater` but I guess that's what the behat tests will highlight once they are available

Reply:
Added some tests, but couldn't go further as indexation is closely related with "status" toggling and that command does not yet support multistore fully. (so it only can switch single shop product status). We could improve tests when that command is made for multistore too, but now its out of scope.